util/accuracy.h

```diff
@@ -49,4 +51,27 @@ inline sycl::half get_ulp_sycl<sycl::half>(sycl::half x) {
   return static_cast<sycl::half>(ulp * multiplier);
 }
 
-#endif // __SYCLCTS_UTIL_ACCURACY_H
+/**
+ * @brief Uses provided allowed difference in ULPs to calculate max allowed
+ *        difference between actual and expected values to consider them equal.
+ *        Returns whether the values are equal by such definition.
+ *        If type T is integral, result of exact comparing is returned.
+ */
+template <typename T>
+bool is_equal_with_ulp(T actual, T expected, unsigned int ulpsExpected) {
+  if (actual == expected) return true;
+  if constexpr (!std::is_integral_v<T>) {
```

Comment:
Non-negative conditions are easier to read.
```suggestion
  if constexpr (std::is_integral_v<T>) return false;
```